### PR TITLE
Add full support for JDBC transaction control methods with Databricks

### PR DESCRIPTION
**Pull request by jayantsing-db**
## Description

Transaction Control APIs:
- setAutoCommit(boolean): Enable/disable auto-commit mode using SET AUTOCOMMIT
- getAutoCommit(): Return cached auto-commit state (JDBC default: true)
- commit(): Commit current transaction using COMMIT SQL command
- rollback(): Rollback current transaction using ROLLBACK SQL command

Error Handling:
- Add DatabricksTransactionException for transaction-specific errors
- Add transaction error codes: TRANSACTION_SET_AUTOCOMMIT_ERROR, TRANSACTION_COMMIT_ERROR, TRANSACTION_ROLLBACK_ERROR
- Preserve original SQLException details (message, SQL state, vendor code)
- Support error code extraction from exception chains

Implementation Details:
- Store autoCommit state in DatabricksSession (initialized to true)
- Honor deprecated ignoreTransactions flag for backward compatibility
- ROLLBACK is forgiving (no-op when no active transaction)
- COMMIT throws exception when no active transaction
- Support REPEATABLE_READ isolation level (Snapshot isolation)
- All transaction methods include comprehensive JavaDoc
- Add new connection property FetchAutoCommitFromServer (default: 0)
- When enabled (=1), getAutoCommit() queries server using "SET AUTOCOMMIT" SQL instead of returning cached value

## Testing
- Add end-to-end integration tests in TransactionTests
- Cover success scenarios (single/multi-table commits, rollbacks)
- Cover failure scenarios (no active transaction, invalid isolation levels)
- Cover edge cases (query failures, UPDATE/DELETE, atomicity)
- Cover multi-table transactions (atomicity, MERGE, repeatable reads)
- All test tables use catalog-owned Delta tables for MST compatibility

### CODE REVIEW

**Comment by jayantsing-db** [comment]
>Please try to have smaller PRs, which are non blocking to the release workflow

This is test heavy PR.

File: src/main/java/com/databricks/jdbc/api/impl/DatabricksConnection.java [hunk 1]
```diff
@@ -777,4 +1014,22 @@ private void fetchCurrentSchemaAndCatalog() throws DatabricksSQLException {
...
+   * masking the original exception.
+   *
+   * @param statement the statement to close, may be null
+   */
+  private void closeStatementSafely(Statement statement) {
+    if (statement != null) {
+      try {
+        statement.close();
+      } catch (SQLException e) {
+        LOGGER.warn("Failed to close statement", e);
```
[the remark this answers is not in the document]
Comment (jayantsing-db) [reply src/main/java/com/databricks/jdbc/api/impl/DatabricksConnection.java]: No this should be an error log. Done

File: src/main/java/com/databricks/jdbc/api/impl/DatabricksDatabaseMetaData.java [hunk 2]
```diff
@@ -827,21 +827,21 @@ public int getMaxUserNameLength() throws SQLException {
   public int getDefaultTransactionIsolation() throws SQLException {
     LOGGER.debug("public int getDefaultTransactionIsolation()");
     throwExceptionIfConnectionIsClosed();
-    return Connection.TRANSACTION_READ_UNCOMMITTED;
+    return Connection.TRANSACTION_REPEATABLE_READ;
```
[the remark this answers is not in the document]
Comment (jayantsing-db) [reply src/main/java/com/databricks/jdbc/api/impl/DatabricksDatabaseMetaData.java]: MST weren't supported by Databricks till now and this was a placeholder minimum isolation level to prevent this API from throwing error.

With Databricks supporting MST (PrPr), we will have to make this change to communicate the correct isolation level which non-traditional repeatable read (SS). One of the reasons, I will request team to consider bumping up a major version (3.1.0).

[the remark this answers is not in the document]
Comment (jayantsing-db) [reply src/main/java/com/databricks/jdbc/api/impl/DatabricksDatabaseMetaData.java]: Now that i think about this, I will use the connection param to communicate the old isolation level

**Review by gopalldb — APPROVED** [review]
